install/install.go

```diff
@@ -659,6 +659,14 @@ func (k *K8sInstaller) preinstall(ctx context.Context) error {
 			}
 		}
 	}
+
+	if k.params.Encryption == encryptionIPsec {
+		// TODO(aanm) automate this as well in form of helm chart
+		if err := k.createEncryptionSecret(ctx); err != nil {
+			return err
+		}
+	}
+
```

Comment:
let me push back a bit on carrying this logic forward.

in general, i'd like to get to the point where cilium-cli does not create any additional resources that are not managed by helm. otherwise we'll end up in a similar situation as the classic mode where resources are managed in 2 different places.

my proposal would be to do either:

- address this todo `// TODO(aanm) automate this as well in form of helm chart` and put the logic in the helm chart if we think this resource should be managed by cilium.

or

- keep the logic as is to make it clear that the user is responsible for managing this resource.

Reply:
Very good remark

I would opt to place the secret creation inside the helm chart. However the generation of the secret data itself would be hard as as far as I know helm has no provision for key generation? Would adding a data field to the values be an option that when it is set it will create the secret?
We then would have to store the key inside the values, would that be a potential security risk?